opensearch/examples/index_template.rs

```diff
@@ -0,0 +1,223 @@
+        .put_index_template(IndicesPutIndexTemplateParts::Name("books"))
+        .body(json!({
+            "index_patterns": ["books-*"],
+            "priority": 0,
+            "template": {
+                "settings": {
+                    "index": {
+                        "number_of_shards": 3,
+                        "number_of_replicas": 0
+                    }
+                }
+            }
+        }))
+        .send()
+        .await?;
+
+    client
+        .indices()
+        .put_index_template(IndicesPutIndexTemplateParts::Name("books-fiction"))
+        .body(json!({
+            "index_patterns": ["books-fiction-*"],
+            "priority": 1,
+            "template": {
+                "settings": {
+                    "index": {
+                        "number_of_shards": 1,
+                        "number_of_replicas": 1
+                    }
+                }
+            }
+        }))
+        .send()
+        .await?;
+
+    // When we create an index named `books-fiction-romance`, OpenSearch will apply the `books-fiction-*` template's settings to the index:
+
+    client
+        .indices()
+        .create(IndicesCreateParts::Index("books-fiction-romance"))
+        .send()
+        .await?;
+    client
+        .indices()
+        .get(IndicesGetParts::Index(&["books-fiction-romance"]))
+        .send()
+        .await?;
+
+    // Composable index templates are a new type of index template that allow you to define multiple component templates and compose them into a final template. The following example creates a component template named `books_mappings` with default mappings for indices of the `books-*` and `books-fiction-*` patterns:
+
+    client
+        .cluster()
+        .put_component_template(ClusterPutComponentTemplateParts::Name("books_mappings"))
+        .body(json!({
+            "template": {
+                "mappings": {
+                    "properties": {
+                        "title": { "type": "text" },
+                        "author": { "type": "text" },
+                        "published_on": { "type": "date" },
+                        "pages": { "type": "integer" }
+                    }
+                }
+            }
+        }))
+        .send()
+        .await?;
+
+    client
+        .indices()
+        .put_index_template(IndicesPutIndexTemplateParts::Name("books"))
+        .body(json!({
+            "index_patterns": ["books-*"],
+            "composed_of": ["books_mappings"],
+            "priority": 0,
+            "template": {
+                "settings": {
+                    "index": {
+                        "number_of_shards": 3,
+                        "number_of_replicas": 0
+                    }
+                }
+            }
+        }))
+        .send()
+        .await?;
+
+    client
+        .indices()
+        .put_index_template(IndicesPutIndexTemplateParts::Name("books"))
+        .body(json!({
+            "index_patterns": ["books-*"],
```

Comment:
This should be the `books-fiction` template
```suggestion
        .put_index_template(IndicesPutIndexTemplateParts::Name("books-fiction"))
        .body(json!({
            "index_patterns": ["books-fiction-*"],
```